File: src/main/java/com/sleepkqq/sololeveling/proto/config/DefaultGrpcClientConfig.java
package com.sleepkqq.sololeveling.proto.config;

import io.grpc.ManagedChannel;
import io.grpc.ManagedChannelBuilder;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public abstract class DefaultGrpcClientConfig {

  private final String host;

  public ManagedChannel createManagedChannel(int port) {
    return ManagedChannelBuilder.forAddress(host, port)
        .usePlaintext()
        .build();
  }
}

File: src/main/kotlin/com/sleepkqq/sololeveling/proto/config/DefaultGrpcClientConfig.kt
package com.sleepkqq.sololeveling.proto.config

import io.grpc.ManagedChannel
import io.grpc.ManagedChannelBuilder

abstract class DefaultGrpcClientConfig {

	fun createManagedChannel(host: String, port: Int): ManagedChannel =
		ManagedChannelBuilder.forAddress(host, port)
			.usePlaintext()
			.build()
}
